lxd/instance/drivers/qmp/commands.go

```diff
@@ -597,3 +597,15 @@ func (m *Monitor) AddSecret(id string, secret string) error {
 
 	return nil
 }
+
+// HasCommand returns true if the given command is available and the feature is enabled.
+// A command might be listed in `query-commands`, but when called return an error like
+//
+//	"The feature '<cmd>' is not enabled"
+//
+// In this case, HasCommand will return false.
+func (m *Monitor) HasCommand(cmd string) bool {
```

Comment:
I think this function is misleading as it actually runs the command in question, and if it didn't take arguments, or did something destructive like "stop" then it might cause unexpected behaviour. I suggest as this function isn't large to just make the desired QMP calls within the context of where you're using this function.